### ISSUE TITLE
[BUG] Add-PnPPageWebPart Cannot Set Images for ImageGallery 

### ISSUE THREAD

**Issue by leahann** [pnp/powershell#3415]
### Reporting an Issue or Missing Feature
Very similar to some earlier bugs related to the Image web part, I'm unable to add images to the Image Gallery web part.  I can add an Image Gallery web part that pulls dynamically from a library, but our use case requires select images in a carousel.  

### Expected behavior 
Images appear in the web part

### Actual behavior 
Spots for the images appear in the web part, but the images aren't visible.
![image](https://github.com/pnp/powershell/assets/12057727/24be112c-6641-463e-bbb0-645e3ad35e25)

### Steps to reproduce behavior
I've tried both Add-PnPPageWebPart and Set-PnPPageWebPart, with the same results.

```
Add-PnPPageWebPart -Page $pageTitle -DefaultWebPartType ImageGallery -Section 1 -Column 1 -WebPartProperties $wpPropJson
```

Where images json in the $wpPropJson variable is like below. The galleryImgId looks like it might be generated when the images are added manually from the front end. I've tried a few different variations on this. 

```
"images": [
          {  
            "siteId": "18e1477c-5056-4853-8951-15268c006a7f",  
            "webId": "d1628f6a-d25e-4641-94cd-2de258f3b1dc",  
            "listId": "{C2DF79CF-273B-4225-9DC4-86A907D46E2F}",  
            "id": "35707cf1-ef0d-49ae-abdd-72f2567a9051",  
            "title": "",  
            "isInList": true,  
            "description": "",  
            "altText": "",  
            "galleryImgId": "cccf24be-a78e-486b-ae1c-84c2296be446"  
          },  
          {  
            "siteId": "18e1477c-5056-4853-8951-15268c006a7f",  
            "webId": "d1628f6a-d25e-4641-94cd-2de258f3b1dc",  
            "listId": "{C2DF79CF-273B-4225-9DC4-86A907D46E2F}",  
            "id": "59d95cd0-47fe-42b7-803b-38cfbea2ffd2",  
            "title": "",  
            "isInList": true,  
            "description": "",  
            "altText": "",  
            "galleryImgId": "8af97428-7f0f-42b7-a1db-0f934623e52c"
          }  
        ]
```

### What is the version of the Cmdlet module you are running?
2.2.0

### Which operating system/environment are you running PnP PowerShell on?
- [x] Windows
- [ ] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by ishaisagi-hns** [comment]
Agreed - I have been testing this for a few days and running into the same issue that the image webpart has when you try to apply the images to it using pnp. 
SharePoint does not seem to like us specifying the siteid, webid, fileid using code - the webpart gets added to the page but displays nothing.  if I switch to carousel mode, the controls say there are two images in the webpart, but displaying nothing for the images. 

if you manually add the same images to the webpart using the web interface, they get added and looking at the propertiesjson there is no difference between the ones set by code and the ones manually added, so it must be some hidden property on the webpart that we are not seeing that tells the webpart if the images were validated or not.


here is an example of the "images" property for a gallery with 2 images (pointing to the same image) added by code:

altText      : 
isInList     : True
galleryImgId : 3dc15aae-2fd8-4570-9aad-7e04a91da0a5
siteId       : 220a0a97-1672-46f2-846d-06caef639e80
webId        : 9e6f6675-2a8d-4ef9-9961-2f8cb53e40a2
title        : 
id           : 5092bc1c-769a-4e85-aa87-1384b0f22042
listId       : {A5163959-BAC7-43B3-AA07-722E83DC9F2F}
description  : 

altText      : 
isInList     : True
galleryImgId : 1ab1aac0-28af-4eb6-b21d-de84ca53d945
siteId       : 220a0a97-1672-46f2-846d-06caef639e80
webId        : 9e6f6675-2a8d-4ef9-9961-2f8cb53e40a2
title        : 
id           : 5092bc1c-769a-4e85-aa87-1384b0f22042
listId       : {A5163959-BAC7-43B3-AA07-722E83DC9F2F}
description  : 

and here is the value of the "images" property of that same webpart after I manually (using the UI) add the same image 

siteId         : 220a0a97-1672-46f2-846d-06caef639e80
webId          : 9e6f6675-2a8d-4ef9-9961-2f8cb53e40a2
listId         : {A5163959-BAC7-43B3-AA07-722E83DC9F2F}
id             : 5092bc1c-769a-4e85-aa87-1384b0f22042
title          : 
isInList       : True
description    : 
altText        : 
imageTileWidth : 600
galleryImgId   : 07225691-b523-4baa-bce6-cddbab190c9b

altText        : 
isInList       : True
galleryImgId   : 3dc15aae-2fd8-4570-9aad-7e04a91da0a5
siteId         : 220a0a97-1672-46f2-846d-06caef639e80
webId          : 9e6f6675-2a8d-4ef9-9961-2f8cb53e40a2
title          : 
id             : 5092bc1c-769a-4e85-aa87-1384b0f22042
listId         : {A5163959-BAC7-43B3-AA07-722E83DC9F2F}
description    : 
imageTileWidth : 600

altText        : 
isInList       : True
galleryImgId   : 1ab1aac0-28af-4eb6-b21d-de84ca53d945
siteId         : 220a0a97-1672-46f2-846d-06caef639e80
webId          : 9e6f6675-2a8d-4ef9-9961-2f8cb53e40a2
title          : 
id             : 5092bc1c-769a-4e85-aa87-1384b0f22042
listId         : {A5163959-BAC7-43B3-AA07-722E83DC9F2F}
description    : 
imageTileWidth : 600


you can see that all properties are exactly the same except the galleryImgId (which is random for each image).
